src/ApiClient.php

```diff
@@ -52,35 +57,72 @@ public function search(string $searchTerm): array
     }
 
     /**
-     * Get historical data for a symbol.
+     * Get historical data for a symbol (depreciated).
+     *
+     * @deprecated In future versions, this function will be removed. Please consider using getHistoricalQuoteData() instead.
      *
      * @return array|HistoricalData[]
      *
      * @throws ApiException
      */
     public function getHistoricalData(string $symbol, string $interval, \DateTimeInterface $startDate, \DateTimeInterface $endDate): array
     {
-        $allowedIntervals = [self::INTERVAL_1_DAY, self::INTERVAL_1_WEEK, self::INTERVAL_1_MONTH];
-        if (!\in_array($interval, $allowedIntervals)) {
-            throw new \InvalidArgumentException(sprintf('Interval must be one of: %s', implode(', ', $allowedIntervals)));
-        }
-
-        if ($startDate > $endDate) {
-            throw new \InvalidArgumentException('Start date must be before end date');
-        }
+        error_log('[scheb/yahoo-finance-api] getHistoricalData() is deprecated and will be removed in a future release');
```

Comment:
Let's use this:
```php
@trigger_error('[scheb/yahoo-finance-api] getHistoricalData() is deprecated and will be removed in a future release', \E_USER_DEPRECATED);
```

That gives users some more control on this error, e.g. completely ignoring `E_USER_DEPRECATED` messages.